pkg/services/backfill.go

```diff
@@ -0,0 +1,130 @@
+package services
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"go.uber.org/zap"
+
+	"github.com/google/uuid"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/apperrors"
+	"github.com/cmsgov/easi-app/pkg/models"
+)
+
+// NewBackfill imports historical data into EASi
+func NewBackfill(
+	config Config,
+	fetchIntake func(ctx context.Context, id uuid.UUID) (*models.SystemIntake, error),
+	fetchIntakeByLifecycleID func(ctx context.Context, lcid string) (*models.SystemIntake, error),
+	createIntake func(c context.Context, intake *models.SystemIntake) (*models.SystemIntake, error),
+	updateIntake func(ctx context.Context, intake *models.SystemIntake) (*models.SystemIntake, error),
+	createNote func(c context.Context, note *models.Note) (*models.Note, error),
+	authorize func(context.Context) (bool, error),
+) func(context.Context, models.SystemIntake, []models.Note) (bool, error) {
+	return func(ctx context.Context, intake models.SystemIntake, notes []models.Note) (bool, error) {
+		ok, err := authorize(ctx)
+		if err != nil {
+			return false, err
+		}
+
+		fmt.Printf("%+v", intake.LifecycleID)
+
+		if intake.LifecycleID.String == "" {
+			// skip it
+			return false, nil
+		}
+
+		if !ok {
+			return false, &apperrors.ResourceNotFoundError{
+				Err:      errors.New("failed to authorize backfill creation"),
+				Resource: models.Note{},
+			}
+		}
+
+		if intake.LifecycleID.ValueOrZero() == "" {
+			return false, &apperrors.BadRequestError{
+				Err: errors.New("lifecycle ID is required"),
+			}
+		}
+
+		existingRecord, existingErr := fetchIntakeByLifecycleID(ctx, intake.LifecycleID.String)
+		if existingErr != nil {
+			return false, &apperrors.QueryError{
+				Err: existingErr,
+			}
+		}
+
+		if existingRecord != nil {
+			appcontext.ZLogger(ctx).Info("updating existing system intake", zap.String("lcid", intake.LifecycleID.String))
+			intake.RequestType = models.SystemIntakeRequestTypeNEW
+			intake.Status = models.SystemIntakeStatusLCIDISSUED
+			intake.ID = existingRecord.ID
+			if _, updateErr := updateIntake(ctx, &intake); updateErr != nil {
+				return false, err
+			}
+			return false, nil
+		}
+
+		appcontext.ZLogger(ctx).Info("creating new system intake", zap.String("lcid", intake.LifecycleID.String))
+		// invariant data for all backfill (maybe do this in transport layer, for de-normalizing fields?)
+		intake.RequestType = models.SystemIntakeRequestTypeNEW // TODO: correct RequestType?
```

Comment:
Does anything need to happen in this PR for this TODO?

Reply:
I don't think so. That was there from the earlier version of the script and I didn't change it.

Reply:
I took out the comment after considering this a bit.